client/src/components/ResultItem/ResultItem.js

```diff
@@ -1,13 +1,34 @@
+/* eslint-disable no-eval */
 import './ResultItem.css';
 
 import { Highlight } from 'react-instantsearch-dom';
 import PropTypes from 'prop-types';
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 
 /* Algolia's Hit: display a result item from search */
 function ResultItem(props) {
   // a hit (result) object that contains all info about a result item
   const { hit } = props;
+  const history = useHistory();
+
+  const handleAnalysis = () => {
+    const { data } = hit;
+    const ad = {
+      headline: data.headline,
+      headlineAnalysis: {
+        sentiment: JSON.parse(data.headlineSentiment),
+        // use eval() due to single-quote appearances in JSON
+        entities: eval(data.headlineTerms),
```

Comment:
Eval is technically not allowed per google js style guide. 
https://google.github.io/styleguide/jsguide.html#disallowed-features-dynamic-code-evaluation

I'll leave it to you to decide whether or not an alternative exists, but you should make note of no alternatives existing if that's the case in your comment for transparency .